libs/base/System/Directory.idr

```diff
@@ -83,7 +84,9 @@ currentDir
     = do res <- primIO prim__currentDir
          if prim__nullPtr res /= 0
             then pure Nothing
```

Comment:
Should there be freeing in the unhappy path too?

Reply:
In the unhappy path the pointer is null, so it doesn't need to be freed. (The man page says that free does nothing if passed null, so we could safely call it if we want.)